cmd/shell.go

```diff
@@ -260,10 +260,47 @@ type CAConfig struct {
 }
 
 // PAConfig specifies how a policy authority should connect to its
-// database, and what policies it should enforce.
+// database, what policies it should enforce, and what challenges
+// it should offer.
 type PAConfig struct {
 	DBConnect              string
 	EnforcePolicyWhitelist bool
+	Challenges             map[string]bool
+}
+
+// UnmarshalJSON is really actually vanilla, but with some validity checks and
```

Comment:
There's no need to define a custom UnmarshalJSON for all of PAConfig: instead you could do `type ChallengesOffered *map[string]bool` and then define a custom UnmarshalJSON for that.

However, that said: I think it would probably be better to stick with the vanilla unmarshal for that too, and do the validation / default setting in NewPolicyAuthorityImpl.